### PR TITLE
Add beta integration test for V1 release

### PR DESCRIPTION
**Pull request by JeromeJu**
# Changes
This commit adds the `pull-tekton-pipelines-beta-integration-test`

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by JeromeJu** [comment]
cc @dibyom

**Review by dibyom — APPROVED** [review]

File: prow/config.yaml [hunk 1]
```diff
@@ -1354,6 +1354,66 @@ presubmits:
...
+              cpu: 3500m
+              memory: 4Gi
+            limits:
+              cpu: 3500m
+              memory: 8Gi
+          env:
+            - name: PIPELINE_FEATURE_GATE
+              value: "beta"
+            - name: EMBEDDED_STATUS_GATE
+              value: "minimal"
```
Comment (chitrangpatel) [inline prow/config.yaml:1416]: (nit) Probably not just true here but also for other integration tests. 

Do we explicitly need to set  `PIPELINE_FEATURE_GATE` and `EMBEDDED_STATUS_GATE` env variables here? There are probably others defined in `e2e-tests-kind-prow-beta.env` that are likely being set by the runner script any way?

Comment (JeromeJu) [reply prow/config.yaml:1416]: Good call. This is actually a design question @dibyom and I had in some conversation. Do we seek a way to refactor ATM or maybe in a follow-up issue/ PR?

File: prow/config.yaml [hunk 2]
```diff
@@ -1354,6 +1354,66 @@ presubmits:
...
+            - "--" # end kubernetes_execute_bazel flags (consider following flags as text)
+            - "/usr/local/bin/kind-e2e"
+            - "--k8s-version"
+            - "v1.23.x"
+            - "--nodes"
+            - "3"
+            - "--e2e-script"
+            - "./test/e2e-tests.sh"
+            - --e2e-env
+            - "./test/e2e-tests-kind-prow-beta.env"
```
Comment (afrittoli) [inline prow/config.yaml:1402]: This file does not exists in the pipeline repo

Comment (afrittoli) [reply prow/config.yaml:1402]: We should wait until https://github.com/tektoncd/pipeline/pull/5737 is merged

**Review by chitrangpatel — APPROVED** [review]

**Review by vdemeester — APPROVED** [review]
/kind misc

File: prow/config.yaml (hunk 1 above)
Comment (dibyom) [reply prow/config.yaml:1416]: @JeromeJu I think we can just remove the env fields that are also present in the env file in pipelines

Comment (JeromeJu) [reply prow/config.yaml]: Thanks @dibyom , updated.

**Comment by chitrangpatel** [comment]
https://github.com/tektoncd/pipeline/pull/5737 is merged
/unhold

**Pull request merged**